hypertrace-trace-enricher/enriched-span-constants/src/main/java/org/hypertrace/traceenricher/enrichedspan/constants/utils/EnrichedSpanUtils.java

```diff
@@ -374,4 +416,212 @@ public static boolean areBothSpansFromDifferentService(Event event, Event parent
+
+    // If the rpc system is grpc then read the keys with prefix "rpc.response.metadata"
+    if (isRpcSystemGrpc(attributes)) {
+      responseMetadataMap.putAll(
+          attributes.entrySet().stream()
+              .filter(entry -> entry.getKey().startsWith(RPC_RESPONSE_METADATA_PREFIX))
+              .filter(entry -> nullCheck(entry.getValue()))
+              .collect(
+                  Collectors.toUnmodifiableMap(
+                      entry -> entry.getKey().substring(RPC_RESPONSE_METADATA_PREFIX.length()),
+                      entry -> entry.getValue().getValue())));
+    }
+
+    return Collections.unmodifiableMap(responseMetadataMap);
+  }
+
+  /**
+   * The request cookies are populated as `http.request.header.cookie` with value as
+   * cookie1=value1;cookie2=value2;cookie3=value3
+   *
+   * @return map of cookie key -> value
+   *     <ul>
+   *       <li>cookie1 -> value1
+   *       <li>cookie2 -> value2
+   *       <li>cookie3 -> value3
+   *     </ul>
+   */
+  public static Map<String, String> getRequestCookies(Event event) {
+    String requestCookie = getStringAttribute(event, REQUEST_COOKIE_HEADER_KEY);
+    if (requestCookie == null || requestCookie.isEmpty()) {
+      return Collections.emptyMap();
+    }
+
+    Map<String, String> cookies = new HashMap<>();
+
+    List<String> cookiePairs = SEMICOLON_SPLITTER.splitToList(requestCookie);
+    for (String cookiePair : cookiePairs) {
+      List<String> cookieKeyValue = COOKIE_KEY_VALUE_SPLITTER.splitToList(cookiePair);
+      if (cookieKeyValue.size() != 2) {
+        LOGGER.debug(
+            "Invalid cookie pair {} for tenant {} and event {}",
+            cookiePair,
+            event.getCustomerId(),
+            HexUtils.getHex(event.getEventId()));
+        continue;
+      }
+
+      cookies.put(cookieKeyValue.get(0), cookieKeyValue.get(1));
+    }
+
+    return Collections.unmodifiableMap(cookies);
+  }
+
+  /**
+   * The response cookies are populated as `http.response.header.set-cookie[0] -> cookie1=value1`,
+   * `http.response.header.set-cookie[1] -> cookie2=value2` and so on in span attributes
+   *
+   * @return map of cookie key -> value *
+   *     <ul>
+   *       *
+   *       <li>cookie1 -> value1
+   *       <li>cookie2 -> value2
+   *     </ul>
+   */
+  public static Map<String, String> getResponseCookies(Event event) {
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Collections.emptyMap();
+    }
+
+    List<HttpCookie> cookies = new ArrayList<>();
+
+    Map<String, AttributeValue> attributes = event.getAttributes().getAttributeMap();
+    for (Map.Entry<String, AttributeValue> entry : attributes.entrySet()) {
+      String attributeKey = entry.getKey();
+
+      if (!isHttpResponseCookie(attributeKey)) {
+        continue;
+      }
+
+      String attributeValue = entry.getValue().getValue();
+      try {
+        cookies.addAll(HttpCookie.parse(attributeValue));
```

Comment:
Any reason for reading request cookies by splitting by '=' vs reading response cookies with `parse`? Can they be unified?